src/town_cmd.cpp

```diff
@@ -3576,7 +3576,7 @@ TownActions GetMaskOfTownActions(CompanyID cid, const Town *t)
 			const TownActions cur = (TownActions)(1 << i);
 
 			/* Is the company not able to bribe ? */
-			if (cur == TACT_BRIBE && (!_settings_game.economy.bribe || t->ratings[cid] >= RATING_BRIBE_MAXIMUM)) continue;
+			if (cur == TACT_BRIBE && (!_settings_game.economy.bribe || (t->ratings[cid] >= RATING_BRIBE_MAXIMUM && (t->exclusivity == _current_company || t->exclusivity == INVALID_COMPANY)))) continue;
```

Comment:
4 `))))` is probably indicating that the condition is getting too complex. Split into multiple ifs?